lib/action_subscriber/message_retry.rb

```diff
@@ -0,0 +1,38 @@
+module ActionSubscriber
+  module MessageRetry
+    def self.redeliver_message_with_backoff(env)
+      ttl = calculate_ttl(env)
+      return unless ttl
+      with_exchange(env, ttl) do |exchange|
+        exchange.publish(env.encoded_payload,
+          :content_type => env.content_type,
+          :routing_key => env.routing_key,
+          :expiration => ttl
+        )
+      end
+    end
+
+    # Private Implementation
+    def self.calculate_ttl(env)
+      death = env.headers.fetch("x-death", [{"original-expiration" => "20"}]).first
+      next_ttl = death["original-expiration"].to_s.to_i * 5
```

Comment:
this `to_s` is because march_hare is currently passing the list of headers as `JavaCharacterArray`s. So I had to convert to a ruby string before calling `to_i`, but really that conversion should be happening when I build the `Middleware::Env` object.